### PR TITLE
Fix issue#512

### LINKED ISSUE #512: Create a separate module for flag list view
Currently, the flag related activities are part of main module only.
Move these to a new module - `flag-listview`

screenshot 2018-10-18 at 4 51 17 pm

The flag related views are used in : 
- World clock
- currency converter

### Why are we doing this?
This module will be hosted on maven as a new separate library.

### PR DESCRIPTION
**Pull request by blongho**
## Enables correct loading and populating the flags of countries in CurrencyConverter and WorldClock 

This fix requires one dependency to be included in build.gradle 
```gradle
// World Country flag
implementation 'com.github.blongho:world-country-data:v1.1.0'
```
Demo of the UI change is [here](https://github.com/blongho/Travel-Mate/blob/fix-issue%23512/screenshots/demo-flag-list.gif)

Fixes #512

## Type of change
Just put an x in the [] which are valid.
- [x ] Bug fix (non-breaking change which fixes an issue)
- [ x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x ] `./gradlew assembleDebug assembleRelease`
- [x ] `./gradlew checkstyle`

# Checklist:
- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 1]
```diff
@@ -37,7 +37,9 @@
 import org.json.JSONObject;
 
 import java.io.IOException;
+import java.util.ArrayList;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: why add these imports? These are not being used

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/UtilitiesFragment.java [hunk 2]
```diff
@@ -60,6 +63,10 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         if (!hasAccelerometer || !hasMagneticSensor) {
             this.mHasMagneticSensor = false;
         }
+        String[] timeZones = TimeZone.getAvailableIDs();
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/UtilitiesFragment.java]: i dont think this is being used anywhere. You can remove this file from commit

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java [hunk 3]
```diff
@@ -40,6 +42,7 @@
     //private DigitalClock mDigitalClock;
 
     private static TextClock mDigitalClock;
+    private static final String mTag = "WorldClockActivity";
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: remove changes from here

File: Android/app/src/main/java/utils/Utils.java [hunk 4]
```diff
@@ -65,12 +66,12 @@ public static String decryt(String d) throws UnsupportedEncodingException {
     }
 
     public static boolean isNetworkConnected(Context context) {
-        ConnectivityManager cm =
-                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
```
Comment (Swati4star) [inline Android/app/src/main/java/utils/Utils.java]: no need of changes in this file also

Comment (blongho) [reply Android/app/src/main/java/utils/Utils.java]: This is my IDE doing auto adjust when i changed line wrap. Fixed this!.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java (hunk 3 above)
Comment (blongho) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: I used this during debugging but forgot to remove after i finished. Now removed.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/UtilitiesFragment.java (hunk 2 above)
Comment (blongho) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/UtilitiesFragment.java]: Used this to see what TimeZone.getAvailableIDs() returned so i can better understand what to modify. Forgot to remove when don. Now removed.

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 1 above)
Comment (blongho) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: I had some debugging functions inside main activity. 

I tested the library to see if country flags populated correctly. Forgot to optimize the imports after i removed the code. Now fixed.

**Comment by blongho** [comment]
@Swati4star, all recommended changes have been made and pushed the branch. 

Do i have to make a different pull request or this suffices?

File: Android/app/google-services.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (Swati4star) [inline Android/app/google-services.json]: just asking, what is the need of this file?

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 6]
```diff
@@ -171,6 +171,7 @@ protected void onCreate(Bundle savedInstanceState) {
                     break;
             }
         }
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: revert this change too

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java [hunk 7]
```diff
@@ -71,6 +72,7 @@ public void onResume() {
 
                 setClock();
                 Log.d("Key", timezone_short);
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: revert this file also

File: Android/app/src/main/java/utils/Utils.java [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import android.util.Base64;
 import android.view.View;
 import android.view.inputmethod.InputMethodManager;
+import android.widget.Toast;
```
Comment (Swati4star) [inline Android/app/src/main/java/utils/Utils.java]: revert this

File: Android/app/google-services.json (hunk 5 above)
Comment (blongho) [reply Android/app/google-services.json]: Have no idea why and how it got there. Git just included it as i did a ``` git add . ```

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java (hunk 7 above)
Comment (blongho) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: ok, done

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 6 above)
Comment (blongho) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: ok, done

**Comment by blongho** [comment]
@Swati4star, requested changes have been made

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 6 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: not yet reverted. This file is still here

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java [hunk 9]
```diff
@@ -23,6 +23,7 @@
 import io.github.project_travel_mate.R;
 import utils.CurrencyConverterGlobal;
 
+
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: remove this file also

**Comment by blongho** [comment]
@Swati4star i guess this should be done now. I also allowed "edits from maintainers" so please if there is any trivial change like empty line, could you change that?

**Comment by Swati4star** [comment]
sorry for taking this long.

**Comment by blongho** [comment]
> sorry for taking this long.

It is ok. I am glad i could make a valuable contribution to the project

**Pull request merged**